utopia-remix/app/routes/v1.github.authentication.finish.tsx

```diff
@@ -0,0 +1,10 @@
+import { LoaderFunctionArgs } from "@remix-run/node";
+import { handle, handleOptions } from "../util/api.server";
+import { proxy } from "../util/proxy.server";
+
+export async function loader(args: LoaderFunctionArgs) {
+  return handle(args.request, {
+    OPTIONS: handleOptions,
+    GET: proxy,
+  });
+}
```

Comment:
since this exact `loader` is present in all the "just proxy to Haskell" endpoints, shouldn't we use something with even less code repetition? 

is something like this possible?

```
import { proxyToHaskellBackendLoader } from "../util/proxy.server";
export const loader = proxyToHaskellBackendLoader
```

Reply:
Yeah I also thought about making it a little less verbose (we _always_ need an `OPTIONS` loader because of CORS), but that loader is not always present, some proxy the `GET` methods (like this one) as well.
The reason for it is that since we're going route-by-route (and route-by-method for the handlers) it makes sense, imo, to be explicit in what methods we're proxying and what we are not.

We can of course use another function to which you only pass the methods you want to proxy, but it would be basically the same as using the `handle` function directly, with yet-another-layer of indirection 🤔

Reply:
> and route-by-method for the handlers

can I challenge this? I think it would be very confusing if we change GET to handled by remix and keep the POST method for the haskell server

Reply:
I don't follow 🤔 we're not changing anything here, the method checks have two purposes:
1. you are not allowed to call a different method than the exact one we allow in haskell
2. if we need to customize/have additional endpoints/etc. we can do something like this, which imo is very readable

```ts
return handle(req, () => {
  GET: listDogs(),
  POST: createDog(),
  PUT: updateDog(),
  DELETE: deleteDog(),
}
```